src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -704,7 +704,30 @@ namespace cryptonote
       rx_slow_hash(main_height, seed_height, hash.data, bd.data(), bd.size(), res.data, seed_hash ? 0 : miners, !!seed_hash);
     } else {
       const int pow_variant = b.major_version >= 7 ? b.major_version - 6 : 0;
-      crypto::cn_slow_hash(bd.data(), bd.size(), res, pow_variant, height);
+      
+      // Caching pattern declaration
+      using key_type = std::tuple<blobdata, int, uint64_t>;
+      const key_type key = std::make_tuple(bd, pow_variant, height);
+      static std::map<key_type, crypto::hash> cache_blobdata_2_hash;
+      // Caching pattern declaration ends
```

Comment:
I envision, that the pattern could be used as a template of a map with a key_type and value_type as <K, V>, and accepting a lambda, used to generate the value. The question is, whether it's acceptable to use the static map. I fear of thread safety, when the map is used simultaneously, but against this, the template could arrange a scoped lock.

In case you don't know, if your compiler supports it (it should in 2021), the C++11 standard

Reply:
Perhaps you have got other thoughts to share, before I go all-in with this methodology?

Reply:
When using monerod normally, are there also cache hits? That is something to consider. If there are always cache misses (outside core tests) then this might not be a good idea.

Reply:
I'm currently mostly into reducing the testing time. Helping the normal operation wouldn't hurt, so I setup an experiment for this, but I don't see this code being called at all (whether cache be hit or missed). At least not while synchronizing.